src/states/joining_peer.rs

```diff
@@ -41,54 +40,35 @@ pub const BOOTSTRAP_TIMEOUT: Duration = Duration::from_secs(20);
 /// Time after which an attempt to joining a section is cancelled (and possibly retried).
 pub const JOIN_TIMEOUT: Duration = Duration::from_secs(600);
 
-pub struct JoiningPeerDetails {
-    pub network_service: NetworkService,
-    pub full_id: FullId,
-    pub network_cfg: NetworkParams,
-    pub timer: Timer,
-    pub rng: MainRng,
-}
-
-// State of a node while joining a section
+// State of a node after bootstrapping, while joining a section
 pub struct JoiningPeer {
+    core: Core,
     stage: Stage,
-    network_service: NetworkService,
-    msg_filter: MessageFilter,
-    full_id: FullId,
-    timer: Timer,
-    rng: MainRng,
     network_cfg: NetworkParams,
 }
 
 impl JoiningPeer {
-    pub fn new(mut details: JoiningPeerDetails) -> Self {
-        details.network_service.service_mut().bootstrap();
+    pub fn new(mut core: Core, network_cfg: NetworkParams) -> Self {
```

Comment:
I used to dislike excessive abbreviations but I learned to tolerate them if they are common enough (like cfg or msg). 

Regarding `NetworkParams` - we can't call it `NetworkConfig` because that name is already taken - it's the config passed to the quic-p2p library. What I would do is to rename the `network_cfg` variable to `network_params` just to avoid confusion. That said, a better name for `NetworkParams` would be great. Problem is the word "network" is overloaded. Sometimes we use to refer to the underlying TCP/UDP layer and other times to refer to the whole SAFE network. Would be great to come up with a convention to disambiguate these.

Reply:
> but I learned to tolerate them if they are common enough (like cfg or msg)

Yep. I do the evaluation in two steps; 
Is it common enough among people (like cfg and msg)? and secondly 
Is it common enough in the _code_?

The concept of message is in almost every file, **many** times :). So, it is bound to be known and it will certainly decrease cognitive overhead to use `msg`.
`Cfg` though is just a couple of occurrences perhaps? Could just as well spell it out.
Well, that is my thinking process regarding that anyway.

Reply:
Seems @maqi had some suggestions on `NetworkParams` and `NetworkConfig`.

Reply:
Yeah, I'm leaning towards that as well. So `TransportConfig` and `NetworkParams` it is.